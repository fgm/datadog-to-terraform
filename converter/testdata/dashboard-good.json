{
  "title": "Title",
  "description": "Description",
  "restricted_roles": ["7e4f0109-a7e6-4254-b3fe-ff22b63efae3", "681eee93-3844-40cc-a06f-b674b001abbd"],
  "widgets": [
    {
      "id": 6094222782053076,
      "definition": {
        "title": "ECS",
        "type": "group",
        "layout_type": "ordered",
        "widgets": [
          {
            "id": 2061680356847030,
            "definition": {
              "title": "Num Services Running",
              "show_legend": false,
              "legend_size": "0",
              "type": "timeseries",
              "requests": [
                {
                  "q": "sum:aws.ecs.service.running{$env,$client} by {servicename}",
                  "on_right_yaxis": false,
                  "style": {
                    "palette": "dog_classic",
                    "line_type": "solid",
                    "line_width": "normal"
                  },
                  "display_type": "area",
                  "response_format": "event_list",
                  "formulas": [{ "formula": "query1" }, { "formula": "query2" }],
                  "queries": [
                    {
                      "data_source": "metrics",
                      "name": "query1",
                      "query": "avg:system.disk.free{*}.rollup(avg, 60)"
                    },
                    {
                      "data_source": "metrics",
                      "name": "query1",
                      "query": "avg:system.disk.free{*}.rollup(avg, 60)"
                    }
                  ]
                }
              ],
              "yaxis": {
                "scale": "linear",
                "include_zero": true,
                "min": "auto",
                "max": "auto"
              }
            }
          }
        ]
      }
    }
  ],
  "template_variables": [
    { "name": "env", "default": "*", "prefix": "env" },
    { "name": "client", "default": "*", "prefix": "client" }
  ],
  "layout_type": "ordered",
  "is_read_only": true,
  "notify_list": [],
  "reflow_type": "auto",
  "id": "dru-pjw-8zd"
}
