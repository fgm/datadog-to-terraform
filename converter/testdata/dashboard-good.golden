resource "datadog_dashboard" "dashboard-good" {
description = "Description"
is_read_only = true
layout_type = "ordered"
notify_list = []
reflow_type = "auto"
restricted_roles = ["7e4f0109-a7e6-4254-b3fe-ff22b63efae3","681eee93-3844-40cc-a06f-b674b001abbd"]
template_variable {
default = "*"
name = "env"
prefix = "env"
}
template_variable {
default = "*"
name = "client"
prefix = "client"
}
title = "Title"
widget {
group_definition {
layout_type = "ordered"
title = "ECS"
widget {
timeseries_definition {
legend_size = "0"
request {
display_type = "area"
formula {
formula_expression  = "query1"
}
formula {
formula_expression  = "query2"
}
on_right_yaxis = false
q = "sum:aws.ecs.service.running{$env,$client} by {servicename}"
query {
  metric_query {
data_source = "metrics"
name = "query1"
query = "avg:system.disk.free{*}.rollup(avg, 60)"
}
}
query {
  metric_query {
data_source = "metrics"
name = "query1"
query = "avg:system.disk.free{*}.rollup(avg, 60)"
}
}
response_format = "event_list"
style {
line_type = "solid"
line_width = "normal"
palette = "dog_classic"
}
}
show_legend = false
title = "Num Services Running"
yaxis {
include_zero = true
max = "auto"
min = "auto"
scale = "linear"
}
}
}
}
}
}
